tensorboard/plugins/npmi/npmi_demo.py

```diff
@@ -0,0 +1,703 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Sample data exhibiting npmi values."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os.path
+
+from absl import app
+import tensorflow as tf
+import numpy as np
+from tensorboard.plugins.npmi import summary
+
+tf.compat.v1.enable_eager_execution()
+
+# Directory into which to write tensorboard data.
+LOGDIR = "/tmp/npmi_demo"
+MOUNTAINS = [
```

Comment:
would it be possible for this to come from some other data file? It is a bit lengthy and distracting.

Reply:
Completely agree, just didnt know how you would like this to be. Should this be another file I read from e.g. txt or something like that or a python file just holding this var?

Reply:
Putting it into a separate text file is reasonable. In code you could specify that the input file is a list of entities, one entity per line.

Reply:
I see you already went with the text file, but it might be a bit easier to just create an `npmi_demo_data.py` file with this as a variable.  It's slightly less clean in the sense that it's really more plain data than python, but getting the correct path to a data file (especially that works both open source and internally) is a little finicky.  The current approach depends on the current working directory being the Bazel root, and while that's often the case it's a bit unfortunate to have the script fail if it isn't run from there.

AFAIK the existing precedent we have to get a path reference to it that works both externally and internally is this hacky line here:
https://github.com/tensorflow/tensorboard/blob/7511300d6ca5d25d332459d89e5725a47baf225b/tensorboard/program.py#L99-L101

Reply:
Changed to python file for that reason.